test/e2e/zero-ex/v4/compat.ts

```diff
@@ -0,0 +1,74 @@
+import { ethers } from "hardhat";
+
+interface DeployFunction {
+  encode(bytecode: string, args?: []): string;
+}
+
+class CompatInterface extends ethers.utils.Interface {
+  public readonly deployFunction: DeployFunction;
+  constructor(fragments: unknown) {
+    super(fragments as string);
+    this.deployFunction = {
+      encode: (b: string, a?: unknown[]) =>
+        ethers.utils.hexConcat([b, this.encodeDeploy(a || [])]),
+    };
+  }
+}
```

Comment:
I don't quite understand what's going on in this file. What's the purpose of the compatibility interface?
Is it a wrapper to make ethers compatible with the web3 interface, adding all the methods of Web3 that fullMigrateAsync relies on?

Reply:
This is because the deployment code uses `new ethers.utils.Interface` and expects a `deployFunction` which does not exist on the `ethers` that we use ([they use 4.x](https://github.com/0xProject/protocol/blob/96e0ad789913c1e7c46d5a4365e2da4a636aa131/contracts/dev-utils/package.json#L56)). I didn't want to switch versions just for an e2e test, so I just hacked on this function the the exported `Interface` type.

I didn't really write any documentation, or plan on merging this as is. I expect that once they deploy v4 officially, it will be easier to integrate with (like for `v2` and `v3`).